src/js/game.js

```diff
@@ -111,49 +142,68 @@ function Game() {
 			 * vY: y of the nearest vertex
 			 *
 			 */
-			
+
 			var dX = Math.abs(ballX - x - w/2) - w/2,
 				dY = Math.abs(ballY - y - h/2) - h/2,
 				R2 = ballR*ballR,
 				vX, vY, dX2, dY2;
-				
+
 			dX2 = (dX < 0) ? 0 : dX*dX;
 			dY2 = (dY < 0) ? 0 : dY*dY;
-			
+
 			// If collision
-			if (dX2 + dY2 <= R2) {				
+			if (dX2 + dY2 <= R2) {
+				// Fixing bug: move ball outside the brick
+				while (dX2 + dY2 < R2) {
+					ballX	= (ballX - Math.cos(ballTeta)).limitTo(ballR,gameW-ballR);
+					ballY	= (ballY + Math.sin(ballTeta)).limitTo(ballR,gameH-ballR);
+					dX		= Math.abs(ballX - x - w/2) - w/2;
+					dY		= Math.abs(ballY - y - h/2) - h/2;
+					dX2		= (dX < 0) ? 0 : dX*dX;
+					dY2		= (dY < 0) ? 0 : dY*dY;
+				}
+
 				// Check collision for vertical borders, horizontal borders and vertexes
-				if (!dY2) this.newTeta = ballTeta <= pi ? pi - ballTeta : 3*pi - ballTeta;
-				else if (!dX2) this.newTeta =  2*pi - ballTeta;		
+				if (dY < 0) this.tan = Infinity;
+				else if (dX < 0) this.tan =  0;
 				else {
```

Comment:
Using this criteria will drop the limit cases to the vertexes case. This is needed in order to decrease the health of the adjacent brick (if present).